File: 2614-prime-in-diagonal/main.cpp
oclass Solution {
public:
    int diagonalPrime(vector<vector<int>>& nums) {
        
        if (nums.empty()) {
            return 0;
        }

        vector<int> prime_numbers;
        prime_numbers.reserve(nums.size() * 2);

        for (size_t i = 0; i < nums.size(); ++i) {
            prime_numbers.push_back(nums[i][i]);
            prime_numbers.push_back(nums[i][nums.size() - i - 1]);
        }

        int max_value = 0;
        for (int num: prime_numbers) {
            if (is_prime(num) && num > max_value){
                max_value = num;
            }
        }

        return max_value;

    }

private:
    bool is_prime(int num){
        if (num <= 1) {
            return false;
        }

        for (int i = 2; i <= sqrt(num); ++i) {
            if (num % i == 0){
                return false;
            }
        }

        return true;
    }
};

File: 2614-prime-in-diagonal/main.py
class Solution:
    def diagonalPrime(self, nums: List[List[int]]) -> int:
        
        def is_prime(num: int) -> bool:
            
            if num <= 1:
                return False
            
            for i in range(2, int(num ** 0.5) + 1):
                if num % i == 0:
                    return False
            
            return True
        
        prime_numbers = [nums[i][i] for i in range(len(nums))]
        prime_numbers += [nums[i][len(nums) - i - 1] for i in range(len(nums))]

        prime_numbers = [num for num in prime_numbers if is_prime(num)]

        return max(prime_numbers) if prime_numbers else 0
